### PR TITLE
Use policy generation to calculate overall condition.

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
There is a race condition when when a policy is being updated and the status is calculated before, 
Reconcile is reached for an enactment so it will use old enactment status, to fix that this PR saves the policy generation at the enactment and check that when calculating the overall condition so it ignore the enactment status with old policy generation.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/nodenetworkconfigurationpolicy/enactmentstatus/conditions/counter_test.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+var _ = Describe("Enactment condition counter", func() {
+	type EnactmentCounterCase struct {
+		enactmentsToCount nmstatev1alpha1.NodeNetworkConfigurationEnactmentList
+		policyGeneration  int64
+		expectedCount     ConditionCount
+	}
+	DescribeTable("the enactments statuses", func(c EnactmentCounterCase) {
+		obtainedCount := Count(c.enactmentsToCount, c.policyGeneration)
+		Expect(obtainedCount).To(Equal(c.expectedCount))
+		//TODO: Do we also check getters ? available().true(), etc...
```
Comment (phoracek) [inline pkg/controller/nodenetworkconfigurationpolicy/enactmentstatus/conditions/counter_test.go]: Please resolve the TODO or drop it. It is not really actionable or informative

File: pkg/controller/nodenetworkconfigurationpolicy/enactmentstatus/conditions/counter.go [hunk 2]
```diff
@@ -12,13 +12,18 @@ type CountByConditionStatus map[corev1.ConditionStatus]int
 
 type ConditionCount map[nmstatev1alpha1.ConditionType]CountByConditionStatus
 
-func Count(enactments nmstatev1alpha1.NodeNetworkConfigurationEnactmentList) ConditionCount {
+func Count(enactments nmstatev1alpha1.NodeNetworkConfigurationEnactmentList, policyGeneration int64) ConditionCount {
 	conditionCount := ConditionCount{}
 	for _, conditionType := range nmstatev1alpha1.NodeNetworkConfigurationEnactmentConditionTypes {
-		conditionCount[conditionType] = CountByConditionStatus{}
+		conditionCount[conditionType] = CountByConditionStatus{
```
[the remark this answers is not in the document]
Comment (qinqon) [reply pkg/controller/nodenetworkconfigurationpolicy/enactmentstatus/conditions/counter.go]: Important part is the map entries, I have discvoer it with unit test, we where not storing them.

[the remark this answers is not in the document]
Comment (phoracek) [reply pkg/controller/nodenetworkconfigurationpolicy/enactmentstatus/conditions/counter.go]: But how does this help? Compared to the original, it looks like here you jjust explicitly initialize the default values.

**Comment by phoracek** [comment]
/lgtm
/approve

Meh, it was just a nit.

**Pull request merged**